kore/src/Kore/Step/Axiom/EvaluationStrategy.hs

```diff
@@ -234,35 +237,34 @@ applyFirstSimplifierThatWorks
                         ++ show applicationResult
                         )
                     )
-                Monad.when
-                    (not (OrPattern.isFalse orRemainders)
+                if  not (OrPattern.isFalse orRemainders)
                     && not (acceptsMultipleResults multipleResults)
```

Comment:
When the simplification rule applies, both checks are conditioned on `not (acceptsMultipleResults multipleResults)`. I think this part of the code would be clearer if we moved that condition to a guard.

Should we also relax the restriction on returning multiple results? That error will probably be triggered the first time someone writes a simplification rule with a disjunction in the `requires`.

Reply:
> When the simplification rule applies, both checks are conditioned on `not (acceptsMultipleResults multipleResults)`. I think this part of the code would be clearer if we moved that condition to a guard.

I thought of this, but wasn't sure. Thanks for mentioning it!

 
> Should we also relax the restriction on returning multiple results? That error will probably be triggered the first time someone writes a simplification rule with a disjunction in the `requires`.

So far, the policy has been to await for the first time when it happens, and then think about how to deal with it in a clearer context.